static/app/components/events/interfaces/frame/stacktraceLink.tsx

```diff
@@ -1,207 +1,185 @@
-  Event,
-  Frame,
-  Organization,
-  Project,
-  StacktraceLinkResult,
-} from 'sentry/types';
+import type {Event, Frame, StacktraceLinkResult} from 'sentry/types';
 import {StacktraceLinkEvents} from 'sentry/utils/analytics/integrations/stacktraceLinkAnalyticsEvents';
 import {getAnalyicsDataForEvent} from 'sentry/utils/events';
-import handleXhrErrorResponse from 'sentry/utils/handleXhrErrorResponse';
 import {
   getIntegrationIcon,
   trackIntegrationAnalytics,
 } from 'sentry/utils/integrationUtil';
 import {promptIsDismissed} from 'sentry/utils/promptIsDismissed';
-import withOrganization from 'sentry/utils/withOrganization';
-import withProjects from 'sentry/utils/withProjects';
+import {useQuery} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
 
 import {OpenInContainer} from './openInContextLine';
 import StacktraceLinkModal from './stacktraceLinkModal';
 
-type Props = AsyncComponent['props'] & {
+interface StacktraceLinkProps {
   event: Event;
   frame: Frame;
+  /**
+   * The line of code being linked
+   */
   line: string;
-  lineNo: number;
-  organization: Organization;
-  projects: Project[];
-};
-
-type State = AsyncComponent['state'] & {
-  isDismissed: boolean;
-  match: StacktraceLinkResult;
-  promptLoaded: boolean;
-};
+}
 
-class StacktraceLink extends AsyncComponent<Props, State> {
-  get project() {
-    // we can't use the withProject HoC on an the issue page
-    // so we ge around that by using the withProjects HoC
-    // and look up the project from the list
-    const {projects, event} = this.props;
-    return projects.find(project => project.id === event.projectID);
-  }
+export function StacktraceLink({frame, event, line}: StacktraceLinkProps) {
+  const api = useApi();
+  const organization = useOrganization();
+  const {projects} = useProjects();
+  const project = useMemo(
+    () => projects.find(p => p.id === event.projectID),
+    [projects, event]
+  );
+  const [promptDismissed, setPromptDismissed] = useState(false);
+  const prompt = usePromptsCheck('stacktrace_link', organization.id, project?.id);
+
+  const query = {
+    file: frame.filename,
+    platform: event.platform,
+    commitId: event.release?.lastCommit?.id,
+    ...(event.sdk?.name && {sdkName: event.sdk.name}),
+    ...(frame.absPath && {absPath: frame.absPath}),
+    ...(frame.module && {module: frame.module}),
+    ...(frame.package && {package: frame.package}),
+  };
+  const {
+    data: match,
+    isLoading,
+    refetch,
+  } = useQuery<StacktraceLinkResult>([
+    `/projects/${organization.slug}/${project?.slug}/stacktrace-link/`,
+    {query},
+  ]);
+
+  // Update dismissed state when prompt is loaded
+  useEffect(() => {
+    if (prompt.isLoading || !prompt.data?.data) {
+      return;
+    }
 
-  componentDidMount() {
-    this.promptsCheck();
-  }
+    setPromptDismissed(
```

Comment:
Instead of the useEffect, you could do something like this:

```ts
const queryClient = useQueryClient();
const prompt = usePromptsCheck(
  'stacktrace_link',
  organization.id,
  project?.id
);
const promptDismissed = prompt
  ? promptIsDismissed({
      dismissedTime: prompt.data.data.dismissed_ts,
      snoozedTime: prompt.data.data.snoozed_ts,
    })
  : true;

const dismissPrompt = () => {
  promptsUpdate(...);
  queryClient.setQueryData(makeQueryKey(...), { data: { dismissed_ts: 1 }});
}
```

idk what `dismissed_ts` means but it looks like if it's a number it's dismissed?